### ISSUE TITLE
Inet address 127.0.0.1:9042 is not available: [Errno 48] Address already in use

### ISSUE THREAD

**Issue by sthomp** [pcmanus/ccm#87]
After running 'ccm start' I wanted to shutdown the cluster using 'ccm stop' and 'ccm remove'.

Later, when starting a new cluster, I ran 'ccm start' and received the following exception:

Traceback (most recent call last):
  File "/usr/local/bin/ccm", line 68, in <module>
    cmd.run()
  File "/Library/Python/2.7/site-packages/ccmlib/cmds/cluster_cmds.py", line 402, in run
    if self.cluster.start(no_wait=self.options.no_wait, verbose=self.options.verbose, jvm_args=self.options.jvm_args) is None:
  File "/Library/Python/2.7/site-packages/ccmlib/cluster.py", line 213, in start
    p = node.start(update_pid=False, jvm_args=jvm_args)
  File "/Library/Python/2.7/site-packages/ccmlib/node.py", line 312, in start
    common.check_socket_available(itf)
  File "/Library/Python/2.7/site-packages/ccmlib/common.py", line 176, in check_socket_available
    raise UnavailableSocketError("Inet address %s:%s is not available: %s" % (addr, port, msg))
ccmlib.common.UnavailableSocketError: Inet address 127.0.0.1:9042 is not available: [Errno 48] Address already in use

It looks like org.apache.cassandra.service.CassandraDaemon is still running on that port. Is that normal? Should ccm kill that process when I do either 'ccm stop' or 'ccm remove'? Currently, I have to manually kill the process to start the cluster again.

**Comment by mrn-aglic** [comment]
How to fix this for windows?

**Comment by ptnapoleon** [comment]
@mrn-aglic Can you open a new issue? Are you having problems binding to 127.0.0.1 or one of the other loopbacks, 127.0.0.2, 127.0.0.3, etc

**Comment by mrn-aglic** [comment]
When I run ccm start I get this error:

Traceback (most recent call last):
  File "C:\Python27\Scripts\ccm.py", line 72, in <module>
    cmd.run()
  File "C:\Python27\lib\site-packages\ccmlib\cmds\cluster_cmds.py", line 503, in run
    profile_options=profile_options) is None:
  File "C:\Python27\lib\site-packages\ccmlib\cluster.py", line 249, in start
    p = node.start(update_pid=False, jvm_args=jvm_args, profile_options=profile_options)
  File "C:\Python27\lib\site-packages\ccmlib\node.py", line 443, in start
    common.check_socket_available(itf)
  File "C:\Python27\lib\site-packages\ccmlib\common.py", line 341, in check_socket_available
    raise UnavailableSocketError("Inet address %s:%s is not available: %s" % (addr, port, msg))
ccmlib.common.UnavailableSocketError: Inet address 127.0.0.1:9042 is not available: [Errno 10013] An attempt was made to access

**Comment by ptnapoleon** [comment]
@mrn-aglic Then you either already have a Cassandra instance running on that port, or a previously one had an issue shutting down and new nodes are failing to listen on that address. Check for a current process first.

**Comment by mrn-aglic** [comment]
I can't find a cassandra process in task list... How do I check if a previously running instance had an issue shutting down?

**Comment by ptnapoleon** [comment]
Unfortunately, C* only shows up as `java` in tasklist. Do you have a JDK installed? You could use `jps` to find `CassandraDaemon`. 

If the process is gone, but a new one can't bind, that's a different problem. Open a new issue for that, and we'll investigate if it is a Cassandra problem or a CCM one.

**Comment by mrn-aglic** [comment]
When running JPS it outputs a number with JPS. 
Could my error be caused by insufficient memory on the machine?

**Comment by ptnapoleon** [comment]
Okay, so there is definitely not a Cassandra process currently running based on the jps output. Your error does not appear to be insufficient memory.

**Comment by mrn-aglic** [comment]
When installing ccm, I followed this tutorial: http://www.datastax.com/dev/blog/ccm-2-0-and-windows

**Comment by ankit249** [comment]
$ ps -ef | grep ccm
$ kill -9 {pid}

before starting ccm use following config
$ sudo ifconfig lo0 alias 127.0.0.2
$ sudo ifconfig lo0 alias 127.0.0.3
$ sudo ifconfig lo0 alias 127.0.0.4

and then run the ccm commands
$ ccm create -v 3.2.1 -n 4 AnkitCluster
$ ccm start

you might see JDK 1.8 related errors, ignore them 

$ ccm status
Cluster: 'AnkitCluster'
-----------------------
node1: UP
node3: UP
node2: UP
node4: UP

**Comment by pranavanmaru** [comment]
[CENTOS]
after interfaces are created for loopback addresses and brought up, if you're still facing the error,
`sudo netstat -tulpn | grep :9042`
check what process is listening to port 9042
run `kill 'process id'` to shut down that process and run `ccm start` again.

**Comment by singh-vikas** [comment]
> ccm status

Worked
